hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -976,6 +989,12 @@ private void startJoin() {
                 final int count = members.size() - 1 + setJoins.size();
                 final List<Future> calls = new ArrayList<Future>(count);
                 for (MemberInfo member : setJoins) {
+                    if (node.isMaster()) {
+                        //Only master node should send clusterId and startTime
+                        long startTime = clusterClock.getClusterStartTime();
+                        calls.add(invokeClusterOperation(new FinalizeJoinOperation(memberInfos, postJoinOp, time,
```

Comment:
`FinalizeJoinOperation` is sent from only master node and to only newly joined nodes. I don't know how you decided that it's sent by every node to newly joining member. `startJoin()` method is also called by master node only. 

So no need to check master here, no need for `FinalizeJoinOperation.isClusterInfoUpdateOperation()` etc...